core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackValidator.java

```diff
@@ -36,12 +40,28 @@
     @Inject
     private PlatformResourceClientService platformResourceClientService;
 
+    @Inject
+    private EntitlementService entitlementService;
+
     public void validate(Stack subject, ValidationResult.ValidationResultBuilder validationBuilder) {
         if (CollectionUtils.isEmpty(subject.getInstanceGroups())) {
             validationBuilder.error("Stack request must contain instance groups.");
         }
         validateTemplates(subject, validationBuilder);
         validateEncryptionKey(subject, validationBuilder);
+        validateVariant(subject, validationBuilder);
+    }
+
+    private void validateVariant(Stack source, ValidationResultBuilder validationBuilder) {
+        String variant = source.getPlatformVariant();
+        boolean awsNativeEnabled = entitlementService.awsNativeEnabled(source.getCreator().getTenant().getName());
+        if (AWS_NATIVE_VARIANT.variant().value().equals(variant)) {
+            if (awsNativeEnabled) {
```

Comment:
The two conditions can be merged into a single statement, unless some further checks are also needed inside the outer `if` block.